src/main/java/net/minecraftforge/common/IHorseArmor.java

```diff
@@ -0,0 +1,25 @@
+package net.minecraftforge.common;
+
+import net.minecraft.entity.passive.EntityHorse;
+import net.minecraft.entity.passive.HorseArmorType;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+
+public interface IHorseArmor 
+{
```

Comment:
Why is this an interface instead of just getHorseArmorType on item returning null?

Reply:
Initially it was the latter, as my understanding is that Forge has been against using interfaces in this scenario (at least prior to 1.12 requiring Java 8) for 2 reasons: 
1. Adding new methods to an interface breaks binary compat.
2. Calling the interface methods requires an instanceof check and a cast, which is messy.
Are these reasons correct, and are there any more?

1.12 requiring Java 8 solves the first problem with the default modifier. The second problem is not much of an issue in this case as the methods are called in only two places. Since I thought Forge was against using interfaces in this scenario, I didn't use one. Then @liach implied that Forge convention had changed in that regard, so I changed it to an interface because I find that cleaner than shoving more methods in Item. If Forge convention has not changed, then I will change it back.

Reply:
I suggested for interfaces like `IPlantable` and `ISpecialArmor` exist. Move it into item class then.

Reply:
@LexManos It's been two weeks since I replied to your question about why I used an interface. Can I please get an answer?